Write up microsim lit review findings (#1514)
ihmeuw/vivarium_research: docs/source/model_design/vivarium_overview/vivarium_versus_other_model_types/index.rst

~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~

Another complexity enabled by microsimulation is the inclusion of time-varying rates dependent on multiple factors. A good example of this is in our microsimulation of multiple myeloma, 
which allowed relapse and mortality hazard rates to vary over time since treatment line initiation, in addition to being impacted by the line of treatment, demographic covariates, and treatment regimen. 
Read more about how we calculated these rates :ref:`here <calc_mortality_relapse_hazard>`.

The figure below demonstrates how survival rates in multiple myeloma patients differ based on the time since treatment line initiation. This figure was produced by

> researchers at Amgen, Inc, and you can read the paper in which it was published `here <https://www.tandfonline.com/doi/full/10.1080/10428194.2020.1827253>`__. [Braunlin-MM-2020]_ 

Author: This was actually an RST hack to be able to use 'here' as hyperlink text in multiple places on the same page, without getting the duplicate link warning from Sphinx!

Simulant agency
+++++++++++++++

Unlike agent-based models, microsimulations do not necessarily program agency into individual simulants. This removes some of the real-world likeness of these models in favor 
of reflecting realistic population-level dynamics that would be more difficult to program in agency-based simulation. The Vivarium simulations that might be described as having 
the most simulant agency would be our microsimulation of the entire US population, which you can read more about at `<vivarium_census_prl_synth_data>`_, as well as
our simulation of cardiovascular disease, which you can read more about at `<2019_cause_ihd>`_. In the prior simulation, simulants are programmed with propensities to migrate into and out of the 
country, change addresses and employment, have babies, and more. In the latter, simulants are initiated into the population with different adherence rates to medications and likelihoods of discontinuing

> said medications - in turn, this affects their level of risk for moving to a worsened state.

Reviewer: ```suggestion
said medications - in turn, this affects their level of risk for moving to a disease state.
``` 
?

health interventions.

Computational needs
+++++++++++++++++++

To date, all vivarium simulations require supercomputers or high-performance clusters due to their complexity and computational demands. In contrast, frameworks such as the Starsim series focus on 
computational ease. HPVsim, for example, utilizes features that improve computational efficiency and accessibility, allowing simulations to run on a wider range of computing resources, from personal 
computers to cloud-based and HPC systems. Key features of HPVsim include: 

- The “MultiSim” class for managing and analyzing multiple simulations.
- The “Scenario” class for comparing simulations with varying input parameters, including different interventions.
- The “Sweep” class for conducting parameter sweeps and generating heatmaps of outcomes.

These features enable parallel runs of simulations, making it feasible to conduct both small and large-scale simulations without requiring supercomputing resources. Read more about HPVsim
`here <https://www.medrxiv.org/content/10.1101/2023.02.01.23285356v1>`__.

Type of event simulation
++++++++++++++++++++++++

> The way simulants undergo events (i.e.ove from one state to another) can be recorded in different ways, depending on microsimulation methods. These divergent methods are referred 

Reviewer: ```suggestion
The way simulants undergo events (i.e. move from one state to another) can be recorded in different ways, depending on microsimulation methods. These divergent methods are referred 
```

- Can be less computationally efficient compared to discrete event simulation, especially for large-scale models, because it requires evaluating the state of all simulants at each time step, regardless of whether events occur. 

To read more about how we select the length of time steps in our simulations, please see `<vivarium_best_practices_time_steps>`_.

Discrete event simulation 
~~~~~~~~~~~~~~~~~~~~~~~~~

On the other hand, discrete event simulation allows simulants to move from event to event, with the timing of these events determined by sampling from probability distributions. For this reason,
discrete event simulation is synonymous with continuous time simulation; and could also be thought of as using a global clock. For an example of a microsimulation that uses discrete event simulation, 
see the `Starsim series <https://www.ncbi.nlm.nih.gov/books/NBK293948/#:~:text=Discrete%20event%20simulation%20(DES)%20is,life%20process%2C%20facility%20or%20system>`_. As with discrete time simulation,
there are some advantages and disadvantages associated with this type of event simulation.

Advantages include: 

> - Can more accurately model the variability in time that simulants spend in different states, as the timing of transiitons is based on sampled events rather than fixed intervals. 

Reviewer: ```suggestion
- Can more accurately model the variability in time that simulants spend in different states, as the timing of transitions is based on sampled events rather than fixed intervals. 
```

- Can be more computationally efficient for many scenarios, as computations are only performed when events occur, rather than at every time step. This efficiency can be particularly significant in models where events are infrequent relative to the chosen time step in a discrete time simulation. 

Disadvantages include:

> - Requires knowledge of the maximum rate or the sum of all rates leading out of a state to calculate valid transition probability. This can complicate model setup, especially in models with complex state transition dynamics.

Reviewer 2: I'm not sure I completely follow this -- did it come from Starsim?

I think I might say:

```suggestion
- Requires knowledge of how to sample event times, as opposed to event probabilities. This can complicate model setup, especially in models with complex state transition dynamics.
```

estimates, but also other data sources) and typically include population age and sex/gender structure, risk factor exposures and effects, all-cause mortality rates, cause-specific parameters, 
intervention coverages, and more. If any of these checks fail, we debug and go through the process again until model results behave as expected. To read more about our team's V&V process, see 
`<vivarium_best_practices_results_processing>`_.

In contrast, the Framework for Reconstructing Epidemiological Dynamics (FRED) (an agent-based model) has a calibration process that is directly integrated into the simulation's runtime, 
allowing the model to adjust its parameters dynamically to ensure a better fit with empirical data. Unlike the separate V&V process, where models are developed and then retrospectively compared and 
adjusted based on data, FRED's integrated calibration allows for a more dynamic and responsive approach to fitting model outputs to real-world evidence. You can read more about FRED 
`here <https://bmcpublichealth.biomedcentral.com/articles/10.1186/1471-2458-13-940>`__.

.. note:: 

>   Although the authors of FRED do not explicitly name it as a microsimulation, they refer to it as an agent-based modeling system that represents every individual in a sepcific geographic region. 

Reviewer: ```suggestion
  Although the authors of FRED do not explicitly name it as a microsimulation, they refer to it as an agent-based modeling system that represents every individual in a specific geographic region. 
```

>   Their use of a built-in calibration mechanism provides a useful contrast to the retrospective and iterative process utilized by Vivarium researchers.

Reviewer: This auto calibration is something that I think would be super cool to add to Vivarium!

Correlation
+++++++++++

> When developing a Vivarium simulation, we typically overlay population-level statistic generated through the GBD onto individual simulants, rather than using individual-level input data such 

Reviewer 2: ```suggestion
When developing a Vivarium simulation, we typically overlay population-level statistics generated through the GBD onto individual simulants, rather than using individual-level input data such 
```

as hospital records or registry data. This approach involves taking broad, aggregated data and distributing its implications onto the simulated individuals to approximate their health status, risks, and 
progression within the model's context. This modeling strategy is not ubiquitous across all public health microsimulations, nor even all Vivarium microsimulations. For instance, our aforementioned
microsimulation of cardiovascular disease (`<2019_cause_ihd>`_) uses inpatient hospital and claims data to inform the burden of heart failure due to each of the underlying causes of heart failure. These

> data sources include vital registry data from Mexico, Brazil, Taiwan, Colombia, and the US; inpatient admissions from Friuli Venezia, Italy; and linked vital registry data from Friuli Venezia, Italy.

Reviewer: As discussed, putting the link to correlation example from CVD: https://vivarium-research.readthedocs.io/en/latest/models/risk_correlation/sbp_ldlc_fpg_bmi/index.html
